exasol/python_extension_common/deployment/language_container_deployer.py

```diff
@@ -265,10 +267,16 @@ def _check_if_requested_language_alias_already_exists(
                 raise RuntimeError(warning_message)
 
     @classmethod
-    def create(cls, bucketfs_name: str, bucketfs_host: str, bucketfs_port: int,
-               bucketfs_use_https: bool, bucketfs_user: str,
-               bucketfs_password: str, bucket: str, path_in_bucket: str,
+    def create(cls,
                dsn: str, db_user: str, db_password: str, language_alias: str,
+               bucketfs_host: Optional[str] = None, bucketfs_port: Optional[int] = None,
+               bucketfs_name: Optional[str] = None, bucket: Optional[str] = None,
+               bucketfs_user: Optional[str] = None, bucketfs_password: Optional[str] = None,
+               bucketfs_use_https: bool = True,
+               saas_url: Optional[str] = None,
+               saas_account_id: Optional[str] = None, saas_database_id: Optional[str] = None,
+               saas_token: Optional[str] = None,
+               path_in_bucket: str = '',
```

Comment:
Just an idea: Would it be better to have two different class methods: `createOnPrem()` and `createSaas()`?

Reply:
Not sure. The decision of which way to go needs to be taken somewhere. By making two different methods we are pushing this decision to the caller. From the caller's perspective, it might be easier to grab all available parameters, e.g. defined in the config, pass them over to the single point of call and let it make sense of these parameters.

Reply:
I agree that we would move the responsibility to the client. I'am not sure about the client's case. However, if the client knows that the connection is on-prem or SaaS, it would be cumbersome to pass all the unnecessary parameters.

Reply:
Well :), except that the `dsn, db_user and db_password` are not really required for SaaS. They are derivatives of the other SaaS parameters. But we will sort it out in subsequent PRs. Need to first merge the saas-api-python with the helper function that works out the saas connection parameters.

Reply:
Or maybe I should merge it first and then do things property in this PR,